extension/mysql_to_doris/all_tables.sh

```diff
@@ -0,0 +1,142 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+####################################################################
+# This script is used to will mysql databases import doris by external
+####################################################################
+
+#reference configuration file
+source ./conf/mysql.conf
+source ./conf/doris.conf
+
+#define mysql database and doris database
+d_mysql=$1
+d_doris=$2
+
+#check args
+if [ ! -n "$1" ];then
+        echo "please check source database"
+        exit
+fi
+if [ ! -n "$2" ];then
+        echo "please check sink database"
+        exit
+fi
+
+#mkdir files to store tables and tables.sql
+mkdir -p files
+rm -rf ./files/tables
+rm -rf ./files/tables.sql
+
+#get tables from mysql databases
+echo "use $d_mysql; show tables;" |mysql -h$mysql  -uroot -p$mysql_password 2>/dev/null >> ./files/tables
+
+#delete tables first line
+sed -i '1d' ./files/tables
+
+#reference tables to create tables.sql
+for table in $(awk -F '\n' '{print $1}' ./files/tables)
+        do
+        sed -i "/${table}view/d" ./files/tables
+        echo "use $d_mysql; show create table ${table};" |mysql -h$mysql_host -uroot -p$mysql_password 2>/dev/null >> ./files/tables.sql
+        echo "print ${table} sql to tables.sql in the file dir"
+
+done
+
+echo '==============================start to transform mysql table for doris extral table==========================='
+#adjust sql
+awk -F '\t' '{print $2}' ./files/tables.sql |awk '!(NR%2)' |awk '{print $0 ";"}' > ./files/tables1.sql
+sed -i 's/\\n/\n/g' ./files/tables1.sql
+sed -n '/CREATE TABLE/,/ENGINE\=/p' ./files/tables1.sql > ./files/tables2.sql
+#delete tables special struct
+sed -i '/^  CON/d' ./files/tables2.sql
+sed -i '/^  KEY/d' ./files/tables2.sql
+rm -rf ./files/tables.sql
+rm -rf ./files/tables1.sql
+mv ./files/tables2.sql ./files/tables.sql
+
+#start transform tables struct
+sed -i '/ENGINE=/a) ENGINE=MYSQL\n COMMENT "MYSQL"\nPROPERTIES (\n"host" = "ApacheDorisHostIp",\n"port" = "3306",\n"user" = "root",\n"password" = "ApacheDorisHostPassword",\n"database" = "ApacheDorisDataBases",\n"table" = "ApacheDorisTables");' ./files/tables.sql
+
+#delete match line
+sed -i '/ENGINT=/d' ./files/tables.sql
+sed -i '/PRIMARY KEY/d' ./files/tables.sql
+sed -i '/UNIQUE KEY/d' ./files/tables.sql
+#delete , at the beginning (
+sed -i '/,\s*$/{:loop; N; /,\(\s*\|\n\))/! bloop; s/,\s*[\n]\?\s*)/\n)/}' ./files/tables.sql
+
+#delete a line on keyword
+sed -i -e '$!N;/\n.*ENGINE=MYSQL/!P;D' ./files/tables.sql
+#replace mysql password、database、table、host
+for t_name in $(awk -F '\n' '{print $1}' ./files/tables)
+        do
+        sed -i "0,/ApacheDorisHostIp/s/ApacheDorisHostIp/${mysql}/" ./files/tables.sql
```

Comment:
It is incorrect to use ${mysql} here, it should be ${mysql_host}

Reply:
I have completed the repair and test once